dipy/utils/parallel.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import multiprocessing
+from tqdm.auto import tqdm
+from dipy.utils.optpkg import optional_package
+
+joblib, has_joblib, _ = optional_package('joblib')
+dask, has_dask, _ = optional_package('dask')
+
+
+def parfor(func, in_list, out_shape=None, n_jobs=-1, engine="joblib",
+           backend="threading", func_args=[], func_kwargs={},
```

Comment:
`locky` backend should be the default. The other backends do not manage well the threading and we have to manage numpy, openmp, blas, etc global variable

Reply:
Done in b76ef7e2a